src/pages/Diagnostics.vue

```diff
@@ -1,11 +1,27 @@
 <script lang="ts">
+
 import Vue from 'vue';
+import { mapGetters, mapState } from 'vuex';
 
 import DiagnosticsBody from '@/components/DiagnosticsBody.vue';
+import type { ServerState } from '@/main/commandServer/httpCommandServer';
 
 export default Vue.extend({
   name:       'diagnostics',
   components: { DiagnosticsBody },
+
+  data() {
+    return { rows: [] };
+  },
+  async fetch() {
+    await this.$store.dispatch('credentials/fetchCredentials');
+    await this.$store.dispatch('diagnostics/fetchDiagnostics', this.credentials as ServerState);
```

Comment:
See elsewhere — shouldn't need a cast like this.

Reply:
It's needed here. I get a `dispatch deprecated` error without it. I don't know why, and no one else on the internet seems to either.

Reply:
FYI very few people out there would get a `dispatch deprecated` because these notices are entirely of our making. Unfortunately, properties get typed as `any` when using `...mapState('credentials', ['credentials'])`, and this usage is marked as deprecated in  `src/typings/store.d.ts`. 

Vuex has known deficiencies when working with TypeScript and we appear to be running into them. This, and several other reasons, is why the Vue core team recommends that developers look into Pinia moving forward.

@ericpromislow @mook-as